### PR TITLE
[Feature] Add tracking evaluation

### PR DESCRIPTION
**Pull request by zwhus**
This PR should not be merged until the PR https://github.com/open-mmlab/mmdetection/pull/9897 and https://github.com/open-mmlab/mmdetection/pull/9906 is merged.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

This PR should not be merged until the PR #9897 and #9906 is merged.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/evaluation/metrics/coco_video_metric.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                if self._coco_api is None:
+                    assert 'instances' in img_data_sample, \
+                        'ground truth is required for evaluation when ' \
+                        '`ann_file` is not provided'
+                    gt['anns'] = img_data_sample['instances']
+                # add converted result to the results list
+                self.results.append((gt, result))
+
+    def evaluate(self, size: int) -> dict:
+        """Evaluate the model performance of the whole dataset after processing
```
[the remark this answers is not in the document]
Comment (zwhus) [reply mmdet/evaluation/metrics/coco_video_metric.py:82]: it‘s same as the function in BaseVideoMetric, but coco_video_metric inhetrits coco_metric, so it's need to keep

File: mmdet/evaluation/metrics/mot_challenge_metric.py [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+    allowed_benchmarks = ['MOT15', 'MOT16', 'MOT17', 'MOT20', 'DanceTrack']
+    default_prefix: Optional[str] = 'motchallenge-metric'
+
+    def __init__(self,
+                 metric: Union[str, List[str]] = ['HOTA', 'CLEAR', 'Identity'],
+                 outfile_prefix: Optional[str] = None,
+                 track_iou_thr: float = 0.5,
+                 benchmark: str = 'MOT17',
+                 format_only: bool = False,
+                 postprocess_tracklet_cfg: Optional[List[dict]] = [],
```
Comment (JingweiZhang12) [inline mmdet/evaluation/metrics/mot_challenge_metric.py:72]: Please add  docstring about this argument

**Review by JingweiZhang12 — COMMENTED** [review]
Please cherry-pick this commit: https://github.com/JingweiZhang12/mmdetection/commit/57f6cdaf52196ddbd5deac3ca1d13d13eb49e144 and pull the latest code to fix UT.

**Comment by JingweiZhang12** [comment]
Please cherry-pick this commit: https://github.com/JingweiZhang12/mmdetection/commit/57f6cdaf52196ddbd5deac3ca1d13d13eb49e144 since  it's more easy to expand other frame sampling class later after adding `BaseFrameSample`

**Review by JingweiZhang12 — APPROVED** [review]

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**